api/sip/client.py

```diff
@@ -283,6 +284,7 @@ def __init__(
         use_ssl=False,
         ssl_cert=None,
         ssl_key=None,
+        ssl_contexts: Callable[[_SSLMethod], ssl.SSLContext] = ssl.SSLContext,
```

Comment:
This is just for testing. We pass in a function that constructs SSL contexts so that we can mock them easily.